File: app/src/main/java/br/edu/farol/gadoplus/adapter/PesagemAdapter.java
package br.edu.farol.gadoplus.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import br.edu.farol.gadoplus.R;
import br.edu.farol.gadoplus.model.Pesagem;


public class PesagemAdapter extends RecyclerView.Adapter<PesagemAdapter.PesagemViewHolder> {

    private List<Pesagem> pesagems;
    private OnItemClickListener listener;
    private OnItemLongClickListener longListener;

    @Override
    public PesagemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recyclerview_item, parent, false);


        return new PesagemViewHolder(itemView);
    }


    @Override
    public void onBindViewHolder(PesagemViewHolder viewHolder, int i) {
        final Pesagem pesagem = pesagems.get(i);
        if (pesagem != null){
            viewHolder.nome.setText(pesagem.getData());
            viewHolder.descricao.setText(pesagem.getDescricao());



            viewHolder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    listener.onItemClick(pesagem);
                }
            });

            viewHolder.itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View view) {
                    longListener.onItemLongClick(pesagem);
                    return false;
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        if (pesagems != null)
            return pesagems.size();
        else return 0;
    }

    public void setPesagem(List<Pesagem> pesagems) {
        this.pesagems = pesagems;
        notifyDataSetChanged();
    }

    class PesagemViewHolder extends RecyclerView.ViewHolder {

        private TextView nome, descricao, sLote;

        public PesagemViewHolder(View itemView) {
            super(itemView);
            nome = itemView.findViewById(R.id.text_view_title);
            descricao = itemView.findViewById(R.id.text_view_description);
            sLote = itemView.findViewById(R.id.text_view_priority);

        }
    }

    public interface OnItemClickListener {
        void onItemClick(Pesagem pesagem);
    }

    public interface OnItemLongClickListener {
        void onItemLongClick(Pesagem pesagem);
    }


    public void setOnClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }


    public void setOnLongClickListener(OnItemLongClickListener listener) {
        this.longListener = listener;
    }




}

File: app/src/main/java/br/edu/farol/gadoplus/adapter/PesagemAdapter.kt
package br.edu.farol.gadoplus.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.edu.farol.gadoplus.R
import br.edu.farol.gadoplus.adapter.PesagemAdapter.PesagemViewHolder
import br.edu.farol.gadoplus.model.Pesagem

class PesagemAdapter : RecyclerView.Adapter<PesagemViewHolder>() {
    private var pesagems: List<Pesagem>? = null
    private var listener: OnItemClickListener? = null
    private var longListener: OnItemLongClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PesagemViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.recyclerview_item, parent, false)
        return PesagemViewHolder(itemView)
    }

    override fun onBindViewHolder(viewHolder: PesagemViewHolder, i: Int) {
        val pesagem = pesagems!![i]
        if (pesagem != null) {
            viewHolder.nome.text = pesagem.data
            viewHolder.descricao.text = pesagem.descricao
            viewHolder.itemView.setOnClickListener { listener!!.onItemClick(pesagem) }
            viewHolder.itemView.setOnLongClickListener {
                longListener!!.onItemLongClick(pesagem)
                false
            }
        }
    }

    override fun getItemCount(): Int {
        return if (pesagems != null) pesagems!!.size else 0
    }

    fun setPesagem(pesagems: List<Pesagem>?) {
        this.pesagems = pesagems
        notifyDataSetChanged()
    }

    inner class PesagemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nome: TextView
        val descricao: TextView
        private val sLote: TextView

        init {
            nome = itemView.findViewById(R.id.text_view_title)
            descricao = itemView.findViewById(R.id.text_view_description)
            sLote = itemView.findViewById(R.id.text_view_priority)
        }
    }

    interface OnItemClickListener {
        fun onItemClick(pesagem: Pesagem?)
    }

    interface OnItemLongClickListener {
        fun onItemLongClick(pesagem: Pesagem?)
    }

    fun setOnClickListener(listener: OnItemClickListener?) {
        this.listener = listener
    }

    fun setOnLongClickListener(listener: OnItemLongClickListener?) {
        longListener = listener
    }
}